### ISSUE TITLE
Fail if the spec is not a supported version

### ISSUE THREAD

**Issue by JordiPolo** [softprops/openapi#8]
Currently there is support for OpenAPI 2.0.
I think OpenAPI < 2.0 and > 2.0 is not compatible with 2.0 so probably the deserialization should check the version and fail gracefully

**Comment by softprops** [comment]
Good call. I'm thinking to support 3.0 I should start making the interface more version specific so folks can opt into 3.0
